File: 프로그래머스/lv1/김미래/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>
#include <iostream>
using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> s;
    int check[31][31] ={0,};
    
    for(int i=0; i<moves.size(); i++){
        int line = moves[i]-1;
        for(int j=0; j<board[0].size(); j++){
            int now = board[j][line];
            if(now == 0) {
                continue;
            }
            else{
                if(check[j][line] == 0){
                    check[j][line] = 1;
                    if(!s.empty() && s.top() == now ){
                        answer+=2;
                        s.pop();
                        break;
                    }else{
                    s.push(board[j][line]);
                    break;
                    }
                }
            }
        }
    }
    if(answer == 0) return 0;
    else return answer;
}

File: 프로그래머스/lv1/천옥수/크레인인형뽑기게임.py
def solution(board, moves):
    answer = 0
    # moves 순회한다.
    basket = []
    count = 0

    # board[j]의 마지막 원소를 잡아서 넣어야 함
    # board[j][0] ~ board[j][n-1] 순회
    for idx in moves:
        for row in board:
            if row[idx-1] != 0:
                now = row[idx-1]
                row[idx-1] = 0
                if basket and basket[-1] == now:
                    basket.pop()
                    count+= 2
                    break
                else:
                    basket.append(now)
                break

    return count